helper/testhelpers/teststorage/teststorage.go

```diff
@@ -31,7 +31,7 @@ import (
 )
 
 func MakeInmemBackend(t testing.T, logger hclog.Logger) *vault.PhysicalBackendBundle {
-	inm, err := inmem.NewTransactionalInmem(nil, logger)
```

Comment:
I'd probably revert this again and make transactional the default. We could potentially make this a random switch, to test most backends with either, randomly (emitting a log message to identify it)... But this seems unnecessary, given we can show sufficient equivalence with the randomized testing below.